webview-ui/src/index.tsx

```diff
@@ -4,9 +4,17 @@ import App from "./App";
 import "@vscode/codicons/dist/codicon.css";
 import "./index.scss";
 
+let dataSourceURL: string | undefined;
+if (document && document.location && document.location.search) {
+  const maybeDataSourceUrl = new URLSearchParams(document.location.search).get("dataSourceURL");
+  if (maybeDataSourceUrl) {
+    dataSourceURL = decodeURI(maybeDataSourceUrl);
+  }
+}
+
 ReactDOM.render(
   <React.StrictMode>
-    <App />
+    <App dataSourceURL={dataSourceURL} />
```

Comment:
The other meaningful change in this PR: extract the URL from the query parameters and pass into the app.
This runs exactly one time at App startup on page load. If the URL changes via JS, this will not re-run.